Files/EventArguments/LayoutPreferenceEventArgs.cs

```diff
@@ -0,0 +1,17 @@
+﻿using static Files.ViewModels.FolderSettingsViewModel;
+
+namespace Files.EventArguments
+{
+    public class LayoutPreferenceEventArgs
+    {
+        public readonly LayoutPreferences LayoutPreference;
+
```

Comment:
```suggestion

public readonly FolderSettingsViewModel folderSettingsViewModel;

```

Reply:
@gave92 :thinking:

Reply:
I think FolderSettingsViewModel is already passed as the `sender` argument if we want to use it.

Reply:
I'd put it in EventArgs. Mind merging that suggestion?

Reply:
@d2dyno1 @gave92 Want me to commit this suggestion?

Reply:
And use it in the handler:
```cs
e.FolderSettingsViewModel.UpdateLayoutPreferencesForPath(FilesystemViewModel.WorkingDirectory, e.LayoutPreference);
```

Reply:
> And use it in the handler:
> 
> ```cs
> e.FolderSettingsViewModel.UpdateLayoutPreferencesForPath(FilesystemViewModel.WorkingDirectory, e.LayoutPreference);
> ```

Or just use the sender?

Reply:
But what if sender is not `FolderSettingsViewModel`?

Reply:
@d2dyno1 Why wouldn't it be? I thought this event delegate was tied to FolderSettingsViewModel.

Reply:
Ok.. I thought about future proofing but if you think this is better solution I think we should leave it as is

@duke7553

Reply:
@d2dyno1 I just don't want the event argument to be identical to the sender. That event delegate is exclusively used by FolderSettings at the moment, as it's purpose is pretty specific.